### PR TITLE
Parse RegExp in ignoreErrors

### PR DESCRIPTION
**Pull request by gabz75**
For the issue #49, it looks like the ember-cli team won't support this.

See https://github.com/ember-cli/ember-cli/issues/5954#issuecomment-223831128.

Do you mind considering this improvement?

### CODE REVIEW

**Comment by damiencaselli** [comment]
Hello @gabz75, thanks for the PR.

I'd gladly add your contribution to ember-cli, however I'm not much using Ember these days, so tell me if I get this wrong 😄 

Any parameter in the non-string regex form like `/some-pattern/` will be turned to `'/some-useless-pattern/'`, making Raven not to work as expected, right?
If your PR fixes this, then 👍 

Do you think you could please move the `parseRegexErrors` function in a dedicated helper module and add a couple of tests for it? That would be neat!

**Comment by gabz75** [comment]
Hey @damiencaselli that's correct.

I'll try to find some time in the next days to move the logic into a helper and add tests.

Thanks!

File: app/instance-initializers/raven-setup.js [hunk 1]
```diff
@@ -5,6 +5,16 @@ import config from '../config/environment';
 // compatibility.
 const assign = Ember.assign || Ember.merge;
 
+const parseRegexErrors = function parseRegexErrors(ravenOptions) {
+  return Ember.get(ravenOptions, 'ignoreErrors').map((error) => {
+    if (error.startsWith('/') && error.endsWith('/')) {
```
Comment (devinus) [inline app/instance-initializers/raven-setup.js]: This wont match anything but the simplest regexes, for example regexes can have multiple modifiers at the end.

**Comment by devinus** [comment]
Also, Raven doesn't even support regexes itself. https://github.com/getsentry/raven-js/blob/master/src/utils.js#L95

It just turns regexes into a string. So this will turn regexes into strings into regexes into a string.

**Comment by damiencaselli** [comment]
Good catch @devinus. It's interesting to see Raven does not support flags either, since it only extract the `source` property from a `RegExp`.

I really don't know what to think about this regexp problem, but I feel like we are attempting to perform some black magic hehe

We eventually want to mix _RegExp_ and _String_ objects to pass as an argument for Raven initialization, but Ember does not allow but simple objects to its config.

Raven extracts `source` from a _RegExp_ object, and this may be used to trick Sentry.

Why not give Raven some simple objects that would be treated as regexps?  
Could the following trick Raven tests and work out of the box?

```javascript
// config/environment.js

module.exports = function(environment) {
  var ENV = {

    /* config */

    sentry: {
      ravenOptions: {
        ignoreErrors: [
            'SomeError',
            { source: 'this?is\s an error pat{2}ern' }
        ]
    }
  }
}
```

**Comment by gabz75** [comment]
@damiencaselli I just tried `{source: '...'}`, it doesn't work. 

I just pushed the changes for allowing flags (even though Raven.js won't use it) and extracting the helper into its own module.

**Comment by gabz75** [comment]
Ping on this.

Raven does use regex, just not flags. It concats all errors string or regex into one big regex.

See: https://github.com/getsentry/raven-js/blob/master/src/utils.js#L100

**Comment by damiencaselli** [comment]
Thanks for the addition @gabz75 👍

**Comment by kmiyashiro** [comment]
I actually did try `{source: '...'}` and it does appear to work. This is the result of `whitelistUrls: [{ source: 'embercom' }, { source: 'templates-raw' }]`:

screen shot 2017-04-25 at 12 59 51 pm

**Comment by kmiyashiro** [comment]
I'm confused as to what this PR actually does. I see it is assuming the strings are in the format of `'/mypattern/'`, but ember-cli actually converts `RegExp`s into plain Objects, not strings. That means in order for this to work, you have to pass a string of a `RegExp` pattern. If that is the case, how do you differentiate between a `RegExp` and a string that happens to look like a `RegExp` pattern?

**Pull request merged**